docs: document node drain configmap PCP-3323 (#3575)
spectrocloud/librarium: docs/docs-content/clusters/cluster-management/os-patching.md

The following options are available for OS patching:

- [Patch OS on boot](#patch-os-on-boot)
- [Enable scheduled patching](#enable-scheduled-patching)
- [Patch OS on-demand](#perform-on-demand-patching)

You can use a combination of these options to patch the operating system to the latest version. For example, you can
patch OS on boot, set up scheduled patching every month, and also perform on-demand patches at any given point in time.

> Palette provides the option to provide custom node drain configuration, which gives you fine-grained control of drain

Reviewer: ```suggestion
Palette offers the option to provide custom node drain configuration, which gives you fine-grained control of drain
```
Just to not say provide 2x 😃

   | ----------------------- | --------------------------------------- |
   | Last Applied Patch Time | The date and time of the last OS patch. |
   | Patched Version         | The latest patched version.             |

> ## Configure OS Patching Drain Policy

Reviewer: Should we say something about what "drain" means?

> Palette provides the ability to configure the drain policy during on boot, scheduled and on-demand OS patching. This

Reviewer: ```suggestion
Palette provides the ability to configure the drain policy during on-boot, scheduled, and on-demand OS patching. This
```
maybe?

capability can also be useful during node maintenance. Users can specify a node drain policy using a ConfigMap, which
can be applied to deployed clusters using a cluster profile or through `kubectl`.

### Limitations

> - This operation is not available for existing Kubernetes clusters imported into Palette.

Reviewer: ```suggestion
- This operation is not available for existing Kubernetes clusters that have been previously imported into Palette.
```

- This operation is not available for managed Kubernetes Services such as EKS and AKS.
  - For EKS clusters, you can trigger OS updates from Palette. This would request AWS to update cluster node groups to
    the latest patch version.

### Prerequisites

- An active cluster in Palette.

### Enablement

1. Log in to [Palette](https://console.spectrocloud.com).

2. From the left **Main Menu**, click **Profiles**. Then, click **Add Cluster Profile**. The cluster profile creation
   wizard appears.

3. Fill in a name in the **Name** field. Optionally, add a description in the **Description** field. Select the
   **Add-on** box for the **Type**. Click **Next**.

4. Click **Add Manifest**. The manifest editor appears.

5. Add a name in the **Layer Name** field. Then, click **New manifest**. Assign a name to the internal manifest and
   click on the blue button. An empty editor displays on the right side of the screen.

> 6. Paste the snippet below into the empty editor. This snippet uses a ConfigMap to control the drain policy of the
>    cluster. The ConfigMap provides the following drain configuration parameters.

Reviewer: ```suggestion
6. Paste the snippet below into the empty editor. This snippet includes example values within a ConfigMap to demonstrate how to control the drain policy of the
   cluster. The ConfigMap provides the following drain configuration parameters.
```

   | `data.drain.ignoreDaemonSets`         | Flag to indicate whether to ignore DaemonSet managed pods.                                                                                                      | `boolean`            | `false`           |
   | `data.drain.force`                    | Flag to indicate whether to continue if one or more affected pods do not have a declared controller.                                                            | `boolean`            | `false`           |
   | `data.drain.disableEviction`          | Flag to indicate whether to force drain to use delete operations.                                                                                               | `boolean`            | `false`           |
   | `data.drain.skipWaitForDeleteTimeout` | The length of time, in seconds, to wait for pod deletion to complete. If the time elapses, the drain will continue.                                             | `int`                | `0`               |

>    ```yaml

Reviewer: The indentation looks a bit weird :)

Author: It looks correct in the preview no?

   apiVersion: v1
   kind: ConfigMap
   metadata:
   name: palette-agent-config
   namespace: "cluster-{{ .spectro.system.cluster.uid }}"
   data:
   profiling: disable
   feature.workloads: enable
   drain: |
     {
        "timeout": 600000000,
        "gracePeriod": 600,
        "deleteLocalData": true,
        "ignoreDaemonSets": true,
        "force": true,
        "disableEviction": false,
        "skipWaitForDeleteTimeout": 600
     }
   ```

   :::info

>    You can also specify the drain ConfigMap directly on the cluster by using `kubectl`. Download the

Reviewer: ```suggestion
   You can also specify the drain policy ConfigMap directly on the cluster by using `kubectl`. Download the
```

   [Kubeconfig](./kubeconfig.md) file to connect to the cluster and issue the

>    `kubectl create configmap palette-agent-config --namespace <cluster-id>` command. Then, execute the command

Reviewer: Is it `cluster-id` or the namespace name?
